### PR TITLE
Update DeterminisitcSampler to use OTel core Parent/TraceIDRatio samplers

### LINKED ISSUE #132: Use the OTel Sampler Algorithim
**Is your feature request related to a problem? Please describe.**

The current sampler implementation uses the beeline algorithm for sampling. Unfortunately, this is not compatible with the algorithm that OTel uses which means that this distro cannot play nicely with other OTel SDKs.

**Describe the solution you'd like**

- [ ] Update the sampler implementation to delegate to the stock OTel TraceIdRatio sampler.
- [ ] Ensure that the proper `sample_rate` attribute is still set so that Honeycomb can show proper graphs.
- [ ] Research using TraceIdRatio vs ParentBased(root=TraceIdRatio) and decide which to use.

### PR DESCRIPTION
**Pull request by MikeGoldsmith**
## Which problem is this PR solving?
Updates the DeterministicSampler to internally create and use a ParentBased sampler that uses a TraceIdRatioBased Sampler as it's root. The combination of ParentBased and RatioBased samplers ensure that root span's mark the trace as sampled / not sampled based on a 1/X sample rate, then subsequent spans within the same trace use the same sampling decision.

This PR reduces the complexity of our implementation and depends on the core samplers to better support multi-service & beeline / OTel environments.

- Resolves #132 

## Short description of the changes
- Replace custom logic in DeterministicSampler with internally created Otel core ParentBasedSampler and TraceIdRatioBased sampler
- Sets the DeterministicSampler with a sample rate of 1 (always sampler) as the default sampler when configuring the OpenTelemetry SDK
- Update tests to verify behaviour, including:
- - Always on (sample rate of 1 - 1.0 ratio)
- - Always off (sample rate of 0 0.0 ratio)
- - Deterministic (sampler rate of 2 - 0.5 ratio)
- - Ensure SampleRate attribute is returned in SamplingDecision

### CODE REVIEW

File: src/Honeycomb.OpenTelemetry/DeterministicSampler.cs [hunk 1]
```diff
@@ -1,8 +1,5 @@
 using OpenTelemetry.Trace;
-using System;
 using System.Collections.Generic;
-using System.Security.Cryptography;
-using System.Text;
 
 namespace Honeycomb
```
Comment (vreynolds) [inline src/Honeycomb.OpenTelemetry/DeterministicSampler.cs:4]: Is this a legit IDE warning?
```
Namespace does not correspond to file location, should be: 'Honeycomb.OpenTelemetry'
```

File: test/Honeycomb.OpenTelemetry.Tests/DeterministicSamplerTests.cs [hunk 2]
```diff
@@ -10,17 +10,12 @@ public class DeterministicSamplerTests
         [Fact]
```
Comment (vreynolds) [inline test/Honeycomb.OpenTelemetry.Tests/DeterministicSamplerTests.cs:10]: Should we add a test that SampleRate attribute is set?

File: test/Honeycomb.OpenTelemetry.Tests/DeterministicSamplerTests.cs [hunk 3]
```diff
@@ -56,53 +46,37 @@ public void AlwaysSampleRateAlwaysReturnsDropResult()
...
-                name: "Span",
-                kind: ActivityKind.Client
-            );
-
-            var result = sampler.ShouldSample(parameters);
-            Assert.Equal(result.Decision, expectedDecision);
-            Assert.Collection(result.Attributes,
-                item => Assert.Equal(new KeyValuePair<string, object>("sampleRate", sampleRate), item)
-            );
+            Assert.True(count > 25 && count < 100);
```
Comment (vreynolds) [inline test/Honeycomb.OpenTelemetry.Tests/DeterministicSamplerTests.cs:79]: In the Java test, this expectation was between 25 and 75. Any particular reason it's different in .NET?

File: src/Honeycomb.OpenTelemetry/DeterministicSampler.cs (hunk 1 above)
Comment (MikeGoldsmith) [reply src/Honeycomb.OpenTelemetry/DeterministicSampler.cs:4]: Yeah, this should be `Honeycomb.OpenTelemetry` like the others 👍🏻 .

File: test/Honeycomb.OpenTelemetry.Tests/DeterministicSamplerTests.cs (hunk 3 above)
Comment (MikeGoldsmith) [reply test/Honeycomb.OpenTelemetry.Tests/DeterministicSamplerTests.cs:79]: Yup, this should also be > 25 && < 75 👍🏻

File: test/Honeycomb.OpenTelemetry.Tests/DeterministicSamplerTests.cs (hunk 2 above)
Comment (MikeGoldsmith) [reply test/Honeycomb.OpenTelemetry.Tests/DeterministicSamplerTests.cs:10]: I was going to add a specific test for that, but already tested the attributes were present in the other tests. I'll extract into a discrete test so we can test for that specific use case 👍🏻

File: test/Honeycomb.OpenTelemetry.Tests/DeterministicSamplerTests.cs [hunk 4]
```diff
@@ -10,99 +10,84 @@ public class DeterministicSamplerTests
...
+                var result = sampler.ShouldSample(parameters);
                 Assert.Equal(SamplingDecision.Drop, result.Decision);
-                Assert.Collection(result.Attributes,
-                    item => Assert.Equal(new KeyValuePair<string, object>("sampleRate", sampleRate), item)
-                );
             }
         }
 
         [Fact]
         public void AlwaysSampleRateAlwaysReturnsDropResult()
```
Comment (robbkidd) [inline test/Honeycomb.OpenTelemetry.Tests/DeterministicSamplerTests.cs:30]:
```suggestion
        public void AlwaysSampleRateAlwaysReturnsRecordAndSample()
```

**Review by robbkidd — APPROVED** [review]
Would you like a sample?

File: test/Honeycomb.OpenTelemetry.Tests/DeterministicSamplerTests.cs (hunk 4 above)
Comment (cartermp) [reply test/Honeycomb.OpenTelemetry.Tests/DeterministicSamplerTests.cs]: This is why F# is the best .NET language. You can do

```fsharp
[<Fact>]
let ``AlwaysSample rate always returns record and sample``() = ...
```

...and I'm off the soapbox.

**Pull request merged**